ReportingServicesTools/Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -157,39 +224,102 @@ function Set-RsDatabase
             Write-Verbose "Executing database creation script... Complete!"
         }
         #endregion Create Database if necessary
-        
-        #region Configuring Database rights
-        # Step 3 - Generate database rights script
-        Write-Verbose "Generating database rights script..."
-        $isWindowsAccount = ($DatabaseCredentialType -like "Windows") -or ($DatabaseCredentialType -like "ServiceAccount")
-        $result = $rsWmiObject.GenerateDatabaseRightsScript($username, $Name, $IsRemoteDatabaseServer, $isWindowsAccount)
-        if ($result.HRESULT -ne 0)
-        {
-            Write-Verbose "Generating database rights script... Failed!"
-            throw "Failed to generate the database rights script from the report server using WMI. Errorcode: $($result.HRESULT)"
-        }
-        else
-        {
-            $SQLscript = $result.Script
-            Write-Verbose "Generating database rights script... Complete!"
-        }
-        
-        # Step 4 - Run Database rights script
-        Write-Verbose "Executing database rights script..."
-        try
+
+        #region Create Runtime Database User
+        if (-not $IsExistingDatabaseUser)
         {
-            Invoke-Sqlcmd -ServerInstance $DatabaseServerName -Query $SQLscript -ErrorAction Stop
+            # nothing to do if it is a windows user
+            $isWindowsAccount = ($DatabaseCredentialType -like "Windows") -or ($DatabaseCredentialType -like "ServiceAccount")
+            if (-not $isWindowsAccount)
+            {
+                # Step 3 - Generate database user script
+                Write-Verbose "Generating database user creation script..."
```

Comment:
lets not create users in the database on behalf of the user

Reply:
I get that it might be unexpected, but in a cloud environment, with dedicated DBs and individual services and rapidly appearing/disappearing instances, there would be an expectation of automated setup. If my setup account has the power to create a database and grant a user rights, it probably has the power to create a user too.

However, adding the new user as sysadmin was incorrect on my part. 

The flow I am looking at is this:

1. We run this ps as an admin rights user
2. It optionally:
    * creates a database
    * creates a runtime user
    * grants access to the runtime user
3. It gives PBIRS the runtime user credentials

The steps in 2 go together. Giving it sysadmin? That was a bad copy on my part. :-)

Take a look at the latest.

Reply:
I get your idea however the function should have a single responsibility in this case is to set the database for reporting services.
Creating users should be done before calling the script in a explicit step in the caller

Reply:
> the function should have a single responsibility

I quite agree... but then should it be creating the database on the server or granting the database user rights to that database?

Reply:
Is a bit different because the function replicates the functionality of the rsconfig tool which begins with the expectation that the user running the tool have enough permissions in the database instance and that have the right user, what we have seen is that sometimes the user configuring RS is different than the dba, in such case the dba takes cares of creating the logins and providing the right permissions in order to configure RS

Reply:
Yeah, I always found the `RSConfig` tool to be a bit strange. 

It does some admin stuff (create the DB, etc.) but not all of it, and uses the admin rights user (as the runtime user). It is a bit of a mix of too many things.

In the end, if you insist, we can leave it out, but it seems to make sense to be clean about it - admin activities = all necessary admin activities and not one step more - but I like knowing that one admin account call does everything necessary.

Reply:
I thought this through. I am trying to solve two issues at the same time - admin activities and admin credentials. I will yank the user creation from here. Let's get the admin credentials issue fixed, then we can worry about where the rest belongs.